File: Basic Maths/Check for prime number.cpp
class Solution {
public:
    bool isPrime(int n) {
        //your code goes here
        if(n == 1)
        {
            return false;
        }
        for(int i = 2; i * i <= n; i++)
        {
            if(n % i == 0)
            {
                return false;
            }
        }
        return true;

    }
};


int main() {
    int num;
    cout << "Enter a number: ";
    cin >> num;

    Solution sol;
    if (sol.isPrime(num)) {
        cout << num << " is a prime number." << endl;
    } else {
        cout << num << " is not a prime number." << endl;
    }

    return 0;
}

File: Basic Maths/Check for prime number.py
# Check if a Number is Prime

def is_prime(n):
    if n <= 1:
        return False
    i = 2
    while i * i <= n:
        if n % i == 0:
            return False
        i += 1
    return True


if __name__ == "__main__":
    for num in [2, 7, 10, 13, 1]:
        print(f"{num}: {'Prime' if is_prime(num) else 'Not Prime'}")
